src/sparseml/recipe_template/main.py

```diff
@@ -12,17 +12,72 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+from typing import Dict, List, Optional, Union
+
+from torch.nn import Module
+
+
 __all__ = [
     "recipe_template",
 ]
 
+from sparseml.recipe_template.utils import (
+    ModifierBuildInfo,
+    get_pruning_info,
+    get_quantization_info,
+    get_training_info,
+)
+
 
-def recipe_template(*args, **kwargs) -> str:
+def recipe_template(
+    pruning: Union[str, bool] = False,
+    quantization: Union[str, bool] = False,
+    lr_func: str = "linear",
+    mask_type: str = "unstructured",
+    global_sparsity: bool = False,
+    target: str = "vnni",
+    model: Optional[Module] = None,
+    convert_to_md: bool = False,
+):
     """
     Return a valid recipe given specified args and kwargs
 
-    # TODO: fill params
-    :return: A string representing a valid recipe
+    :param pruning: Union[str, bool] flag, if "true" or `True` then pruning
+        info will be added, can also be set to `acdc`, or `gmp`.
+    :param quantization: Union[str, bool] flag, if "true" or `True` then quantization
```

Comment:
```suggestion
    :param quantization: Union[str, bool] flag, if `true` or `True` then quantization
```

nit, inconsistencies